storage/src/vespa/storage/persistence/persistencethread.cpp

```diff
@@ -102,9 +176,20 @@ PersistenceThread::handlePut(api::PutCommand& cmd, MessageTracker::UP tracker)
         return tracker;
     }
 
-    spi::Result response = _spi.put(getBucket(cmd.getDocumentId(), cmd.getBucket()),
-                                    spi::Timestamp(cmd.getTimestamp()), std::move(cmd.getDocument()), tracker->context());
-    tracker->checkForError(response);
+    if (_sequencedExecutor == nullptr) {
+        spi::Result response = _spi.put(getBucket(cmd.getDocumentId(), cmd.getBucket()),
+                                        spi::Timestamp(cmd.getTimestamp()), std::move(cmd.getDocument()),
+                                        tracker->context());
+        tracker->checkForError(response);
+    } else {
+        _spi.putAsync(getBucket(cmd.getDocumentId(), cmd.getBucket()), spi::Timestamp(cmd.getTimestamp()),
+                      std::move(cmd.getDocument()), tracker->context(),
+                      std::make_unique<ResultTaskOperationDone>(*_sequencedExecutor, cmd.getBucketId(),
+                              makeResultTask([tracker = std::move(tracker)](spi::Result::UP response) {
```

Comment:
`tracker->context()` should be extracted as own variable before calling `putAsync`, as eval order is undefined and `makeResultTask` tries to capture it with a `std::move(tracker)`. Possible to move away `tracker` before it's dereferenced.

Reply:
Fixed